src/coreclr/jit/importercalls.cpp

```diff
@@ -3068,9 +3068,10 @@ GenTree* Compiler::impIntrinsic(GenTree*                newobjThis,
                 }
             }
 
-            GenTree* hwintrinsic = impHWIntrinsic(ni, clsHnd, method, sig, mustExpand);
+            bool     importAsCall = false;
+            GenTree* hwintrinsic  = impHWIntrinsic(ni, clsHnd, method, sig, mustExpand, &importAsCall);
 
-            if (mustExpand && (hwintrinsic == nullptr))
+            if (mustExpand && (hwintrinsic == nullptr) && !importAsCall)
```

Comment:
When would we be in a scenario where we're `mustExpand` and `hwintrinsic == nullptr` and `importAsCall` is true?

This path is basically supposed to be ensuring we never end up with infinite recursion and therefore stack overflow and it seems like the new condition would end up allowing that to exist.